common/src/main/cpp/extensions/openxr_fb_hand_tracking_mesh_extension_wrapper.cpp

```diff
@@ -0,0 +1,309 @@
+	hand_mesh.indices = reinterpret_cast<int16_t *>(memalloc(sizeof(int16_t) * hand_mesh.indexCapacityInput));
+
+	result = xrGetHandMeshFB(hand_tracker, &hand_mesh);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("Failed to retreive hand mesh data for OpenXR XR_FB_hand_tracking_mesh extension, error code: ", result);
+		return false;
+	}
+
+	// convert to clockwise winding order to cull correct face side
+	for (int i = 0; i < hand_mesh.indexCapacityInput; i += 3) {
+		SWAP(hand_mesh.indices[i], hand_mesh.indices[i + 2]);
+	}
+
+	hand_mesh_data[p_hand] = hand_mesh;
+
+	return true;
+}
+
+Ref<ArrayMesh> OpenXRFbHandTrackingMeshExtensionWrapper::get_mesh(Hand p_hand) {
+	ERR_FAIL_COND_V_MSG(!is_enabled(), Ref<ArrayMesh>(), "OpenXR extension XR_FB_hand_tracking_mesh is not available");
+
+	XrHandTrackingMeshFB hand_mesh = hand_mesh_data[p_hand];
+
+	PackedVector3Array godot_vertex_positions = PackedVector3Array();
+	PackedVector3Array godot_vertex_normals = PackedVector3Array();
+	PackedVector2Array godot_vertex_uvs = PackedVector2Array();
+	PackedInt32Array godot_bone_indices = PackedInt32Array();
+	PackedFloat32Array godot_bone_weights = PackedFloat32Array();
+	for (int i = 0; i < hand_mesh.vertexCapacityInput; i++) {
+		godot_vertex_positions.push_back(Vector3(hand_mesh.vertexPositions[i].x, hand_mesh.vertexPositions[i].y, hand_mesh.vertexPositions[i].z));
+		godot_vertex_normals.push_back(Vector3(hand_mesh.vertexNormals[i].x, hand_mesh.vertexNormals[i].y, hand_mesh.vertexNormals[i].z));
+		godot_vertex_uvs.push_back(Vector2(hand_mesh.vertexUVs[i].x, hand_mesh.vertexUVs[i].y));
+
+		godot_bone_indices.push_back(hand_mesh.vertexBlendIndices[i].x);
+		godot_bone_indices.push_back(hand_mesh.vertexBlendIndices[i].y);
+		godot_bone_indices.push_back(hand_mesh.vertexBlendIndices[i].z);
+		godot_bone_indices.push_back(hand_mesh.vertexBlendIndices[i].w);
+
+		godot_bone_weights.push_back(hand_mesh.vertexBlendWeights[i].x);
+		godot_bone_weights.push_back(hand_mesh.vertexBlendWeights[i].y);
+		godot_bone_weights.push_back(hand_mesh.vertexBlendWeights[i].z);
+		godot_bone_weights.push_back(hand_mesh.vertexBlendWeights[i].w);
+	}
+
+	PackedInt32Array godot_indices = PackedInt32Array();
+	for (int i = 0; i < hand_mesh.indexCapacityInput; i++) {
+		godot_indices.push_back(hand_mesh.indices[i]);
+	}
+
+	Ref<ArrayMesh> array_mesh;
+	array_mesh.instantiate();
+
+	Array arrays;
+	arrays.resize(Mesh::ARRAY_MAX);
+	arrays[Mesh::ARRAY_VERTEX] = godot_vertex_positions;
+	arrays[Mesh::ARRAY_NORMAL] = godot_vertex_normals;
+	arrays[Mesh::ARRAY_TEX_UV] = godot_vertex_uvs;
+	arrays[Mesh::ARRAY_BONES] = godot_bone_indices;
+	arrays[Mesh::ARRAY_WEIGHTS] = godot_bone_weights;
+	arrays[Mesh::ARRAY_INDEX] = godot_indices;
+
+	array_mesh->add_surface_from_arrays(Mesh::PRIMITIVE_TRIANGLES, arrays);
+	array_mesh->regen_normal_maps();
+
+	return array_mesh;
+}
+
+void OpenXRFbHandTrackingMeshExtensionWrapper::construct_skeleton(Skeleton3D *p_skeleton, Hand p_hand) {
```

Comment:
Seems like the `Skeleton3D` is the _return value_; in this case it should be the second argument, and the variable name should be updated to `r_skeleton`.